### PR TITLE
Update README for OpenShift 4.0 and CI changes

### PR DESCRIPTION
**Pull request by spadgett**
/cc @rhamilto @jcaianirh

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -206,12 +196,33 @@ Run integration tests on an OpenShift cluster:
...
 ```
 This will include the normal k8s CRUD tests and CRUD tests for OpenShift
-resources. It doesn't include ALM tests since it assumes ALM is not
-set up on an OpenShift cluster.
+resources.
+
+#### How the Integration Tests run in CI
+
+The end-to-end tests run against pull requests using [ci-operator](https://github.com/openshift/ci-operator/).
+The tests are defined in [this manifest](https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/console/openshift-console-master-presubmits.yaml)
```
Comment (jcaianirh) [inline README.md]: The tests are generated by github.com/openshift/release/ci-operator/config/openshift/console/openshift-console-master.yaml using ci-operator-prowgen (https://github.com/openshift/ci-operator-prowgen) into https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/console/openshift-console-master-presubmits.yaml

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -22,38 +14,8 @@ This is a rough outline of what a contributor's workflow looks like:
 
 - Create a topic branch from where you want to base your work (usually master).
 - Make commits of logical units.
-- Make sure your commit messages are in the proper format (see below).
 - Push your changes to a topic branch in your fork of the repository.
 - Make sure the tests pass, and add any new tests as appropriate.
 - Submit a pull request to the original repository.
 
-
-### Format of the Commit Message
```
Comment (spadgett) [inline CONTRIBUTING.md]: Since we're haven't been following this convention, I removed it.

**Comment by spadgett** [comment]
Thanks @jcaianirh updated (with a few additional changes)

**Comment by spadgett** [comment]
flake https://jira.coreos.com/browse/CONSOLE-1293

I'll wait until this is tagged to re-run tests to avoid unnecessary runs.

File: README.md [hunk 3]
```diff
@@ -35,55 +33,63 @@ Backend binaries are output to `/bin`.
 
 ### Configure the application
 
-#### OpenShift
+The following instructions assume you have an existing cluster you can connect
+to. OpenShift 4.0 clusters can be installed using the
+[openshift/installer](https://github.com/openshift/installer). More information
```
Comment (joshuawilson) [inline README.md]: You can also use CodeReady Containers for local installs. https://github.com/code-ready/osp4

File: README.md [hunk 4]
```diff
@@ -35,55 +33,63 @@ Backend binaries are output to `/bin`.
...
+oc login -u kubeadmin -p $(cat /path/to/install-dir/auth/kubeadmin-password)
+source ./contrib/oc-environment.sh
+./bin/bridge
 ```
 
-To run bridge locally connected to an OpenShift cluster, create an `OAuthClient` resource with a generated secret and read that secret:
+The console will be running at [localhost:9000](http://localhost:9000):
+
+If you don't have `kubeadmin` access, you can use any user's API token,
+although you will limited to that user's access and might not be able to run
```
Comment (joshuawilson) [inline README.md]: change `you will limited` to `you will be limited`

File: README.md [hunk 5]
```diff
@@ -206,12 +190,33 @@ Run integration tests on an OpenShift cluster:
...
-resources. It doesn't include ALM tests since it assumes ALM is not
-set up on an OpenShift cluster.
+resources.
+
+#### How the Integration Tests run in CI
+
+The end-to-end tests run against pull requests using [ci-operator](https://github.com/openshift/ci-operator/).
+The tests are defined in [this manifest](https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/console/openshift-console-master-presubmits.yaml)
+in the [openshift/release](https://github.com/openshift/release) repo and were generated with [ci-operator-prowgen](https://github.com/openshift/ci-operator-prowgen).
+CI runs the [test-prow-e2e.sh](test-prow-e2e.sh) script, which uses the `e2e` suite defined in [protractor.conf.ts](frontend/integration-tests/protractor.conf.ts).
```
Comment (joshuawilson) [inline README.md]: Do you want this in a new line? Looks like you need another carriage return or 2 spaces at the end of the line above.

**Comment by spadgett** [comment]
@joshuawilson thanks, updated

**Comment by spadgett** [comment]
I added some additional detail to the contributing doc, including linking to Bugzilla bugs.

@jcaianirh another look?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -35,25 +27,51 @@ questions: what changed and why. The subject line should feature the what and
...
 ```
-scripts: add the test-cluster command
+Add the test-cluster command
 
-this uses tmux to setup a test cluster that you can easily kill and
+This uses tmux to setup a test cluster that you can easily kill and
 start for debugging.
+```
+
+Commits that fix a Bugzilla bug should add the bug number like `Bug 12345: ` to
```
Comment (jcaianirh) [inline CONTRIBUTING.md]: Is this data supposed to be in the first line of the commit message or the description text of the opened PR?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -35,25 +27,51 @@ questions: what changed and why. The subject line should feature the what and
...
+Bug 1679272: Validate console can talk to OAuth token URL
+
+Make sure we can successfully talk to the OAuth token URL after
+discovering metadata before marking the console pod as ready.
 
-Fixes #38
+Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1679272
+```
+
+Pull requests that close a GitHub issue should add text like `Closes #123` to
```
Comment (jcaianirh) [inline CONTRIBUTING.md]: should add text to the pr description such as 'Closes #123' ...

**Comment by spadgett** [comment]
@jcaianirh thanks, updated

File: README.md [hunk 8]
```diff
@@ -35,55 +33,66 @@ Backend binaries are output to `/bin`.
...
 
 ```
-oc login -u system:admin
+oc login -u kubeadmin -p $(cat /path/to/install-dir/auth/kubeadmin-password)
+source ./contrib/oc-environment.sh
+./bin/bridge
 ```
 
-To run bridge locally connected to an OpenShift cluster, create an `OAuthClient` resource with a generated secret and read that secret:
+The console will be running at [localhost:9000](http://localhost:9000):
```
Comment (rhamilto) [inline README.md]: `:` at the end should be`.`?

**Comment by spadgett** [comment]
thanks @rhamilto, updated

**Pull request merged**